### PR TITLE
Add release notes for Trino Gateway 12 and related changes 

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Assemble the release notes for Trino Gateway 12 release and adjust rest of docs to version 12

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

## Verification for each pull request

Format: PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

Any dates missing in the list just had no merged PRs.

## 12 Sep 2024
 
* #463 ✅ rn ✅ docs

## 15 Sep 2024

* #469 ✅ rn ✅ docs

## 18 Sep 2024

* #471 ✅ rn ✅ docs

## 20 Sep 2024

* #474 ✅ rn ✅ docs

## 23 Sep 2024

* #470 ✅ rn ✅ docs

## 24 Sep 2024

* #477 ✅ rn ✅ docs
* #479 ✅ rn ✅ docs

## 27 Sep 2024

* #466 ✅ rn ✅ docs
* #488 ✅ rn ✅ docs
* #481 ✅ rn ✅ docs

## 29 Sep 2024

* #494 ✅ rn ✅ docs
* #496 ✅ rn ✅ docs
* #492 ✅ rn ✅ docs
* #495 ✅ rn ✅ docs
* #493 ✅ rn ✅ docs

## 1 Oct 2024

* #483 ✅ rn ✅ docs
* #489 ✅ rn ✅ docs
* #499 ✅ rn ✅ docs
* #500 ✅ rn ✅ docs

## 2 Oct 2024

* #490 ✅ rn ✅ docs
* #503 ✅ rn ✅ docs
* #506 ✅ rn ✅ docs
* #507 ✅ rn ✅ docs
* #508 ✅ rn ✅ docs
* #509 ✅ rn ✅ docs
* #511 ✅ rn ✅ docs
* #512 ✅ rn ✅ docs

## 3 Oct 2024

* #510 ✅ rn ✅ docs
* #513 ✅ rn ✅ docs
* #514 ✅ rn ✅ docs
* #515 ✅ rn ✅ docs
* #517 ✅ rn ✅ docs

## 11 Oct 2024

* #360 ✅ rn ✅ docs

## 15 Oct 2024 

* #522 ✅ rn ✅ docs
* #521 ✅ rn ✅ docs
* #497 ✅ rn ✅ docs

## 17 Oct 2024 

* #505 ✅ rn ✅ docs

## 18 Oct 2024 

* #525 ✅ rn ✅ docs

## 21 Oct 2024 

* #529 ✅ rn ✅ docs
* #518 ✅ rn ✅ docs

## 23 Oct 2024 

* #486 ✅ rn ✅ docs

## 24 Oct 2024 

* #533 ✅ rn ✅ docs
* #524 ✅ rn ✅ docs

## 25 Oct 2024 

* #535 ✅ rn ✅ docs

## 26 Oct 2024 

* #537 ✅ rn ✅ docs

## 30 Oct 2024 

* #425 ✅ rn ✅ docs

## 31 Oct 2024 

* #541 ✅ rn ✅ docs
* #534 ✅ rn ✅ docs

## 1 Nov 2024 

* #531 ✅ rn ✅ docs

### CODE REVIEW

File: docs/release-notes.md [hunk 1]
```diff
@@ -1,5 +1,26 @@
 # Release notes
 
+## Trino Gateway 11 (dd Oct 2024) { id="12" }
```
Comment (ebyhr) [inline docs/release-notes.md:3]: Trino Gateway 11 → Trino Gateway 12

Comment (mosabua) [reply docs/release-notes.md]: Thanks .. fixed .. messed up during rebasing

File: docs/release-notes.md [hunk 2]
```diff
@@ -1,9 +1,52 @@
 # Release notes
 
+## Trino Gateway 12 (dd Oct 2024) { id="12" }
```
Comment (mosabua) [inline docs/release-notes.md:3]: Date tbd ..

File: docs/release-notes.md [hunk 3]
```diff
@@ -1,9 +1,52 @@
...
+
+Changes:
+
+* Add support for parsing `EXECUTE IMMEDIATE` statements for routing.
+  ([#484](https://github.com/trinodb/trino-gateway/issues/484))
+* Add support to set configuration values from environment variables.
+  ([#483](https://github.com/trinodb/trino-gateway/issues/483))
+* Add support to include information about the cluster used for query processing
+  to the response cookie.
+  ([#478](https://github.com/trinodb/trino-gateway/issues/478))
```
Comment (oneonestar) [inline docs/release-notes.md:22]: Should be #465

File: docs/release-notes.md [hunk 4]
```diff
@@ -76,15 +127,32 @@ Container image `trinodb/trino-gateway:10`
...
-Container image `trinodb/trino-gateway:9`
+Artifacts:
+
+* [JAR file gateway-ha-9-jar-with-dependencies.jar](https://repo1.maven.org/maven2/io/trino/gateway/gateway-ha/9/gateway-ha-9-jar-with-dependencies.jar)
+* Container image `trinodb/trino-gateway:9`
+* Source code as
+  [tar.gz](https://github.com/trinodb/trino-gateway/releases/tag/9.tar.gz)
+  or [zip](https://github.com/trinodb/trino-gateway/releases/tag/9.zip)
+* Helm chart `8` in `helm/trino-gateway` of the tagged source code (version not
+  updated)
```
Comment (oneonestar) [inline docs/release-notes.md:138]: This is awkward. May be we just don't mention Helm chart <= 8?

**Review by Chaho12 — APPROVED** [review]
LGTM 👍  
Can we add this too?
- https://github.com/trinodb/trino-gateway/pull/531

**Comment by mosabua** [comment]
> LGTM 👍 Can we add this too?
> 
> * [Fix parsing failure on WITH clauses #531](https://github.com/trinodb/trino-gateway/pull/531)

Yes.. ideally .. after it is merged..

**Review by willmostly — APPROVED** [review]
LGTM

**Review by vishalya — APPROVED** [review]
LGTM

**Pull request merged**